### PR TITLE
Treat tables with 1 col or 1 row as layout tables in Outlook message viewer

### LINKED ISSUE #11430: Suppress layout tables when reading messages in Outlook with UIA for Word enabled
### Is your feature request related to a problem? Please describe.

When using the "Use UI Automation for Word controls" enhanced setting, reading messages in Outlook exposes way more information, and is more performant, than if this setting is off. However, in newsletters, layout tables are often used to place images or headers, or add other eye candy. With UIA for Word enabled, each and every table, nested and otherwise, is reported. This makes reading newsletters a real pain.

### Describe the solution you'd like

NVDA should apply some heuristics when reading messages in Outlook to suppress tables that are clearly for layout. It already has some heuristics to determine layout tables, and if needed, could also take some inspiration for how Firefox determines if a table is a layout table. Especially those tables that only have one column and/or row are clearly not data tables and could be suppressed safely.

### Describe alternatives you've considered

There aren't really any, since turning off UIA for Word makes the reading experience in Outlook far worse because of performance. Such newsletters suddenly take one to two seconds for each press of the DownArrow to read the next line and refresh the braille display. So the only way is to make UIA in Word the default, and if that's the case, also add some heuristics to suppress layout tables when reading.

### Additional context

This came from a discussion with acquaintances who claimed that NVDA was still not ready for most workplace scenarios due to poor outlook support. I did some comparative testing between JAWS and NVDA, and besides the performance improvements I got when I turned on UIA for Word Controls, found this to be a real blocker for most users.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #11430
Replaces #12820 

### Summary of the issue:
Microsoft Outlook uses a Microsoft word document control to present emails when both reading and composing. NVDA is in the process of switching to using UI Automation to access Microsoft Word document controls due to major performance advantages, and also due to Microsoft no longer maintaining the Office object model.
However, in the UI automation implementation exposed by MS Word document controls, it is impossible to identify the difference between a data table and a layout table. Thus, NVDa reports all layout tables when reading, which is extremely annoying, as the majority of HTML emails use many nested layout tables.

### Description of how this pull request fixes the issue:
Tables are now classed as layout tables in MS Word document controls if the app is Outlook, and the table is read-only (Outlook is in message reading view) and the table only has either 1 column or 1 row.

### Testing strategy:
With use UIA in MS Word controls enabled in NVDA's advanced settings:
• Create a table in Microsoft Word and ensure that NVDA still reports it (number of rows / columns, cell's row and column number etc).
• Create a table in Microsoft Outlook composition view and ensure that NVDA still reports it (number of rows / columns, cell's row and column number etc).
• Open an HTML email in Outlook that has layout tables, and ensure that the table is not reported when moving into and through the table's content with the arrow keys.

### Known issues with pull request:
Treating tables with 1 column or 1 row as layout tables catches a good majority of layout tables. However, there will still be a few that will not be (they might have 2 columns say). However, the choice to suppress all tables in #12820 was not popular.

### Change log entries:
Bug fixes
* When reading emails in Outlook and NVDA is accessing the message with UI Automation, certain tables are now marked as layout tables, which means they will no longer be reported by default. (#11430)

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual testing.
- [x] API is compatible with existing addons.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

**Comment by michaelDCurran** [comment]
@leonardder  Your testing of this pr would be most welcome.

**Comment by codeofdusk** [comment]
> certain tables

Why not say "tables containing only one row or column"?

**Review by seanbudd — APPROVED** [review]

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -37,13 +37,17 @@ If you need this functionality please assign a gesture to the appropriate script
 - NVDA no longer treats the value of UIA sliders as always percentage based.
 - Reporting the location of a cell in Microsoft Excel when accessed via UI Automation again works correctly on Windows 11. (#12782)
 - NVDA no longer sets invalid Python locales. (#12753)
+- When reading emails in Outlook  via UI Automation, reporting of tables is now suppressed by default. (#11430)
```
Comment (leonardder) [inline user_docs/en/changes.t2t:40]: I think this one should go away in favour of the latter.

**Review by LeonarddeR — APPROVED** [review]
Works great for me. Data tables are properly read whereas most layout tables I've seen are properly silenced. I think I've seen one table that was incorrectly treated as data table compared to at least 50 where all was ok.

**Pull request merged**